article/app/services/dotcomponents/OnwardsPicker.scala

```diff
@@ -2,19 +2,148 @@ package services.dotcomponents
 
 import agents.CuratedContentAgent
 import common.Edition
-import model.dotcomrendering.OnwardCollectionResponse
+import model.DotcomContentType.{Interactive, Video}
+import model.dotcomrendering.{
+  CSROnwardsCollectionResponse,
+  OnwardCollectionResponse,
+  OnwardsCollection,
+  OnwardsSource,
+  Trail,
+}
 import model.{ArticlePage, ContentFormat}
+import play.api.mvc.RequestHeader
+import views.support.Commercial.isPaidContent
+
+object PopularTags {
+  val allowedTags = Seq(
+    // sport tags
+    "sport/cricket",
+    "sport/rugby-union",
+    "sport/rugbyleague",
+    "sport/formulaone",
+    "sport/tennis",
+    "sport/cycling",
+    "sport/motorsports",
+    "sport/golf",
+    "sport/horse-racing",
+    "sport/boxing",
+    "port/us-sport",
+    "sport/australia-sport",
+    // football tags
+    "football/championsleague",
+    "football/premierleague",
+    "football/championship",
+    "football/europeanfootball",
+    "football/world-cup-2014",
+    // football team tags
+    "football/manchester-united",
+    "football/chelsea",
+    "football/arsenal",
+    "football/manchestercity",
+    "football/tottenham-hotspur",
+    "football/liverpool",
+  )
+}
 
 class OnwardsPicker(curatedContentAgent: CuratedContentAgent) {
-  def forArticle(article: ArticlePage, edition: Edition): Seq[OnwardCollectionResponse] = {
+  def forArticle(article: ArticlePage, edition: Edition)(implicit
+      // It'd be nice to not have this required here, but it's required in `pressedContentToTrail`
+      // and drilled pretty deep from there.
+      request: RequestHeader,
+  ): Seq[OnwardsCollection] = {
     val format = article.article.content.metadata.format.getOrElse(ContentFormat.defaultContentFormat)
+
+    /** This is a stop-gap until we render everything on the server.
+      * It exists as we lose context of the article when Frontend needs to POST to DCR via the onward service.
+      * i.e.
+      *
+      * Frontend => DCR
+      *   { "onwards": { "url": "https://gu-api.com/series/series-id?formatDesign=ArticleDesign&formatTheme=NewsPillar&formatDisplay=StandardDisplay" }}
+      *
+      * DCR => GET https://gu-api.com/url/above => Frontend
+      *
+      * Frontend parseFormatQueryString => POST DCR {
+      *   "format": {
+      *     "design": "ArticleDesign",
+      *     "formatTheme": "NewsPillar",
+      *     "formatDisplay": "StandardDisplay"
+      *   }
+      * }
+      */
+    val formatString = s"formatDesign=${format.design}&formatTheme=${format.theme}&formatDisplay=${format.display}"
```

Comment:
Are we encoding `format` into the url here because we want DCR to send this encoded string from the client to the server?

Reply:
Indeed, that's right, as we won't have context of the format when we hit the onward services.